extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java

```diff
@@ -145,4 +145,10 @@ public class DataSourceJdbcRuntimeConfig {
     @ConfigItem
     public DataSourceJdbcTracingRuntimeConfig tracing = new DataSourceJdbcTracingRuntimeConfig();
 
+    /**
+     * choose agroal DataSourceImplementation
+     */
+    @ConfigItem
+    public Optional<String> dataSourceImplementation = Optional.empty();
```

Comment:
Make it `Optional<DataSourceImplementation>` instead? Then you can get rid of the if-else statements

Reply:
That's great, it scales even better

Reply:
Then I modify the code and upload again.
Thanks!